promises.tex

```diff
@@ -326,7 +326,13 @@ \section{Promises}\label{s:promises-promises}
 
 \noindent
 then the error message won't appear
-because the call to \texttt{fetch} doesn't raise an exception right away.
+because the function \texttt{fetch} is asynchronous. 
```

Comment:
...is asynchronous,
which means that code executes in this order:

1. inside `try`, ask the computer to execute `fetch(url)` at some point in the future
2. since asking this doesn't cause an exception, skip the `catch`
3. then run `fetch(url)`, which fails without anything in place to catch the exception because the `try...catch` has already finished
4. so `console.log` never runs and the error message doesn't appear